src/lib.rs

```diff
@@ -472,25 +511,38 @@ impl Transitions for Dense {
 ///
 /// This can use enormous amounts of memory when there are many patterns,
 /// but matching is very fast.
-#[derive(Clone, Debug)]
-pub struct Sparse(Vec<StateIdx>);
+pub struct Sparse([StateIdx; 256]);
```

Comment:
This is interesting. I don't think I have a problem with this change, but I'm curious, what motivated it?

Reply:
If you look at the `State` type which is the only place that `Sparse` (or `Dense`) is used we can see that `State` is only used inside a `Vec` in `AcAutomaton`. Thus we might as well use a fixed size array here to save an indirection and allocation since we don't need to worry about `Sparse` being moved around a lot.

Reply:
Hmm OK, I think I buy it.